core/ai.py

```diff
@@ -100,36 +111,141 @@ def chat(chat_id, input_message):
+                if complaint:
                     tool_output_array.append(
                         {
                             "tool_call_id": tool.id,
-                            "output": auth_token
+                            "output": complaint["ServiceWrappers"][0]["service"]["serviceRequestId"]
                         }
                     )
                     run = client.beta.threads.runs.submit_tool_outputs(
-                        thread_id=thread.id,
+                        thread_id=thread_id,
                         run_id=run.id,
                         tool_outputs=tool_output_array
                     )
                     run, status = get_run_status(run, client, thread)
 
-                    message = get_assistant_message(client, thread.id)
+                    message = get_assistant_message(client, thread_id)
 
                     history = {
                         "thread_id": thread.id,
                         "run_id": run.id,
                         "status": status,
                     }
-                    history = json.dumps(history)
-                    set_redis(chat_id, history)
+                    set_redis(chat_id, json.dumps(history))
                     return message, history
                 else:
-                    return "Authentication failed", history
-                """
+                    return "Complaint failed", history
+                
+            elif func_name == "search_complaint":
+                complaint = search_complaint(parameters)
+                if complaint:
+                    tool_output_array.append(
+                        {
+                            "tool_call_id": tool.id,
+                            "output": complaint["ServiceWrappers"][0]["service"]["applicationStatus"]
+                        }
+                    )
+                    run = client.beta.threads.runs.submit_tool_outputs(
+                        thread_id=thread.id,
+                        run_id=run.id,
+                        tool_outputs=tool_output_array
+                    )
+                    run, status = get_run_status(run, client, thread)
+
+                    message = get_assistant_message(client, thread_id)
+
+                    history = {
+                        "thread_id": thread_id,
+                        "run_id": run.id,
+                        "status": status,
+                    }
+                    set_redis(chat_id, json.dumps(history))
+                    return message, history
+                else:
+                    return "Complaint not found", history
+                
+    return assistant_message, history
+
+def audio_chat(chat_id, audio_file):
+    input_message = transcribe_audio(audio_file, client)
+    print(f"The input message is : {input_message}")
+    assistant_message, history =  chat(chat_id, input_message)
+    return assistant_message, history
+
+def bhashini_chat(chat_id, input_message):
```

Comment:
@luv-singh-ai why do we need bhashini_chat() function ? We can still use the existing chat() function inside bhashini_text_chat() right ? 

input text in some language -> bashini function for translating to english -> pass translated english message to chat() -> translate response from chat() to particular language 

this is possible right ?